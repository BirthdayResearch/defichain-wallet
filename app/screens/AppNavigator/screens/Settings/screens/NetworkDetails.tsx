import { useNetworkContext } from '@contexts/NetworkContext'
import { RootState } from '@store'
import React from 'react'
import dayjs from 'dayjs'
import { useSelector } from 'react-redux'
import { View } from '../../../../../components'
import { ThemedText, ThemedSectionTitle, ThemedSectionList, ThemedScrollView, ThemedView } from '../../../../../components/themed'
import { tailwind } from '../../../../../tailwind'
import { translate } from '../../../../../translations'

enum NETWORK {
  ConnectedStatus = 'YOU ARE CURRENTLY CONNECTED TO',
  NetworkDetails = 'NETWORK DETAILS'
}

interface StatItemProps {
  label: string
  status: string
  statusIcon?: JSX.Element
  testID: string
}

export function NetworkDetails (): JSX.Element {
  const { network } = useNetworkContext()
  const { connected, count: blockCount, masternodeCount, lastSync } = useSelector((state: RootState) => state.block)
  const syncFormattedDate = (lastSync != null) ? dayjs(lastSync).format('MMM D, h:mm a') : ''

  const statsData = [
    {
      key: NETWORK.ConnectedStatus,
      data: [
        {
          label: translate('screens/NetworkDetails', 'Network'),
          status: network,
          testID: 'network'
        }, {
          label: translate('screens/NetworkDetails', 'Status'),
          status: translate('screens/NetworkDetails', connected ? 'Connected' : 'Disconnected'),
          testID: 'status',
          statusIcon: (
            <View
              style={tailwind(`h-2 w-2 rounded-full ${connected ? 'bg-green-500' : 'bg-red-500'} mr-1.5`)}
              testID='status_icon_network'
            />)
        }
      ]
    }, {
      key: NETWORK.NetworkDetails,
      data: [
        {
          label: translate('screens/NetworkDetails', 'Last synced'),
          status: syncFormattedDate,
          testID: 'last_sync'
        }, {
          label: translate('screens/NetworkDetails', 'Block height'),
          status: blockCount,
          testID: 'block_height'
        }, {
          label: translate('screens/NetworkDetails', 'Total Masternodes'),
          status: masternodeCount,
          testID: 'total_masternodes'
        }
      ]
    }
  ]

  return (
    <ThemedScrollView testID='network_details'>
      <ThemedSectionList
        keyExtractor={(item: StatItemProps, index: number) => `${index}`}
        renderItem={({ item }): JSX.Element => {
         return <StatItemRow {...item} />
        }}
        renderSectionHeader={({ section }) => {
          switch (section.key) {
            case NETWORK.ConnectedStatus:
              return (
                <ThemedSectionTitle
                  testID='network_detail_title'
                  text={translate('screens/NetworkDetails', section.key)}
                />
              )
            case NETWORK.NetworkDetails:
              return (
                <ThemedSectionTitle
                  testID={section.key}
                  text={translate('screens/NetworkDetails', section.key)}
                />
              )
            default:
              return <></>
          }
        }}
        sections={statsData}
        testID='network_detail_screen_list'
      />
    </ThemedScrollView>
  )
}

function StatItemRow ({ label, status, testID, statusIcon }: StatItemProps): JSX.Element {
  return (
    <ThemedView
      dark={tailwind('bg-gray-800 border-b border-gray-700')}
      light={tailwind('bg-white border-b border-gray-200')}
      style={tailwind('flex flex-row p-4 pr-2 items-center justify-between')}
    >
      <ThemedText
        dark={tailwind('text-gray-200')}
        light={tailwind('text-black')}
        style={tailwind('font-medium')}
        testID={`${testID}_label`}
      >
        {label}
      </ThemedText>

      <ThemedView
        dark={tailwind('text-gray-400')}
        light={tailwind('text-gray-600')}
        style={tailwind('flex-row items-center')}
      >
        {statusIcon}
        <ThemedText
          dark={tailwind('text-gray-400')}
          light={tailwind('text-gray-600')}
          style={tailwind('text-sm font-medium text-gray-600')}
          testID={`${testID}_value`}
        >
          {status}
        </ThemedText>
      </ThemedView>
    </ThemedView>
  )
}
