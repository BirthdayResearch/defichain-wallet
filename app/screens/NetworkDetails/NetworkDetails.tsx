import { useNetworkContext } from '@contexts/NetworkContext'
import { RootState } from '@store'
import React from 'react'
import dayjs from 'dayjs'
import { useSelector } from 'react-redux'
import { View } from '../../components'
import { ThemedText, ThemedSectionTitle, ThemedSectionList, ThemedScrollView } from '../../components/themed'
import { tailwind } from '../../tailwind'
import { translate } from '../../translations'

enum NETWORK {
  ConnectedStatus = 'YOU ARE CURRENTLY CONNECTED TO',
  NetworkDetails = 'NETWORK DETAILS'
}

interface StatItemProps {
  label: string
  status: string
  statusIcon?: JSX.Element
  testID: string
}

export function formatTime (date: string): string {
  return dayjs(date).format('MMM D, h:mm a')
}

export const NetworkDetails = (): JSX.Element | null => {
  const { connected, count: blockCount, masterNodeCount, lastSync } = useSelector((state: RootState) => state.block)

  const { network } = useNetworkContext()

  const statsData = [
    {
      key: NETWORK.ConnectedStatus,
      data: [
        {
          label: translate('screens/NetworkDetails', 'Network'),
          status: network,
          testID: 'network'
        }, {
          label: translate('screens/NetworkDetails', 'Status'),
          status: translate('screens/NetworkDetails', connected ? 'Connected' : 'Disconnected'),
          testID: 'status',
          statusIcon: (
            <View
              style={tailwind(`h-2 w-2 rounded-full ${connected ? 'bg-green-500' : 'bg-red-500'} mr-1.5`)}
              testID='status_icon_network'
            />)
        }
      ]
    }, {
      key: NETWORK.NetworkDetails,
      data: [
        {
          label: translate('screens/NetworkDetails', 'Last Sync'),
          status: (lastSync != null) ? formatTime(lastSync) : '',
          testID: 'last_sync'
        }, {
          label: translate('screens/NetworkDetails', 'Block Height'),
          status: blockCount,
          testID: 'block_height'
        }, {
          label: translate('screens/NetworkDetails', 'Total Masternode'),
          status: masterNodeCount,
          testID: 'total_masterenode'
        }
      ]
    }
  ]

  return (
    <ThemedSectionList
      keyExtractor={(item: StatItemProps, index: number) => `${index}`}
      renderItem={({ item }): JSX.Element => {
       return <StatItemRow {...item} />
      }}
      renderSectionHeader={({ section }) => {
        switch (section.key) {
          case NETWORK.ConnectedStatus:
            return (
              <ThemedSectionTitle
                testID='network_detail_title'
                text={translate('screens/NetworkDetails', section.key)}
              />
            )
          case NETWORK.NetworkDetails:
            return (
              <ThemedSectionTitle
                testID={section.key}
                text={translate('screens/NetworkDetails', section.key)}
              />
            )
          default:
            return <></>
        }
      }}
      sections={statsData}
      testID='liquidity_screen_list'
    />
  )
}

function StatItemRow ({ label, status, testID, statusIcon }: StatItemProps): JSX.Element {
  return (
    <ThemedScrollView
      dark={tailwind('bg-gray-800 border-b border-gray-700')}
      light={tailwind('bg-white border-b border-gray-200')}
      contentContainerStyle={tailwind('flex flex-row p-4 pr-2 items-center justify-between')}
    >
      <ThemedText
        dark={tailwind('text-gray-200')}
        light={tailwind('text-black')}
        style={tailwind('font-medium')}
        testID={`${testID}_label`}
      >
        {label}
      </ThemedText>

      <View style={tailwind('flex-row items-center')}>
        {statusIcon}
        <ThemedText
          dark={tailwind('text-gray-400')}
          light={tailwind('text-gray-600')}
          style={tailwind('text-sm font-medium text-gray-600')}
          testID={`${testID}_value`}
        >
          {status}
        </ThemedText>
      </View>
    </ThemedScrollView>
  )
}
