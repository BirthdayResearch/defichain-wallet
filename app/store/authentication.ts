import { createSlice, PayloadAction } from '@reduxjs/toolkit'

export interface Authentication<T> {
  consume: (passphrase: string) => Promise<T> // also serve as passphrase validation logic
  onAuthenticated: (result: T) => Promise<void>
  // OPTIONAL error handler
  // (CRITICAL invalid passphare error is auto handled, include wipe wallet if necessary)
  onError?: (e: Error) => void

  // messages in passcode UI
  // mandatory, default message is for signing transaction
  message: string
  loading: string
}

const initialState: { authentication?: Authentication<any> } = {}

export const authentication = createSlice({
  name: 'authentication',
  initialState,
  reducers: {
    prompt: (state, action: PayloadAction<Authentication<any>>) => {
      state.authentication = action.payload
    },
    dismiss: (state) => {
      delete state.authentication
    }
  }
})
