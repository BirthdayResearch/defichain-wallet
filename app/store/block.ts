import { createSlice, PayloadAction } from '@reduxjs/toolkit'

interface BlockState {
  count?: number
}

const initialState: BlockState = {
  count: undefined
}

export const block = createSlice({
  name: 'block',
  initialState,
  reducers: {
    updateBlock: (state, action: PayloadAction<{ count: number }>) => {
      state.count = action.payload.count
    }
  }
})
