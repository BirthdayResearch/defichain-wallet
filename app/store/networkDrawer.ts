import { CTransactionSegWit } from '@defichain/jellyfish-transaction'
import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit'

export interface Transaction {
  broadcasted: boolean
  signed: CTransactionSegWit
  title?: string
}

export interface NetworkDrawerState {
  transactions: Transaction[]
  height: number
  err?: Error
}

const initialState: NetworkDrawerState = {
  transactions: [],
  height: 49,
  err: undefined
}

export const networkDrawer = createSlice({
  name: 'networkDrawer',
  initialState,
  reducers: {
    setHeight: (state, action: PayloadAction<number>) => {
      state.height = action.payload
    },
    queueTransaction: (state, action: PayloadAction<Transaction>) => {
      state.transactions = [...state.transactions, action.payload]
    },
    closeNetworkDrawer: (state) => {
      state.transactions = []
      state.err = undefined
    },
    setError: (state, action: PayloadAction<Error>) => {
      state.err = action.payload
    },
    popTransaction: (state) => {
      if (state.transactions.length > 1) {
        state.transactions.shift()
        state.transactions = [...state.transactions]
      }
    }
  }
})

export const isDrawerOpenSelector = createSelector([(state: NetworkDrawerState) => state.transactions, (state: NetworkDrawerState) => state.err],
  (transactions, err) => transactions?.length > 0 || err !== undefined)

export const firstTransactionSelector = createSelector((state: NetworkDrawerState) => state.transactions, (transactions) => transactions[0])
