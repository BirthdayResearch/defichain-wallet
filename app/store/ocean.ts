import { CTransactionSegWit } from '@defichain/jellyfish-transaction'
import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit'

export interface OceanTransaction {
  broadcasted: boolean
  signed: CTransactionSegWit
  title?: string
}

export interface OceanState {
  transactions: OceanTransaction[]
  height: number
  err?: Error
}

const initialState: OceanState = {
  transactions: [],
  height: 49,
  err: undefined
}

export const ocean = createSlice({
  name: 'ocean',
  initialState,
  reducers: {
    setHeight: (state, action: PayloadAction<number>) => {
      state.height = action.payload
    },
    queueTransaction: (state, action: PayloadAction<OceanTransaction>) => {
      state.transactions = [...state.transactions, action.payload]
    },
    closeOceanInterface: (state) => {
      state.transactions = []
      state.err = undefined
    },
    setError: (state, action: PayloadAction<Error>) => {
      state.err = action.payload
    },
    popTransaction: (state) => {
      if (state.transactions.length > 1) {
        state.transactions.shift()
        state.transactions = [...state.transactions]
      }
    }
  }
})

export const isDrawerOpenSelector = createSelector([(state: OceanState) => state.transactions, (state: OceanState) => state.err],
  (transactions, err) => transactions?.length > 0 || err !== undefined)

export const firstTransactionSelector = createSelector((state: OceanState) => state.transactions, (transactions) => transactions[0])
