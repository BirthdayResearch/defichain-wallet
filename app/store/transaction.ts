import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { VMTransaction } from '../screens/AppNavigator/screens/Transactions/screens/stateProcessor'

type LoadingState = 'idle' | 'loading' | 'loadingMore' | 'success' | 'background' | 'error'

export interface TransactionState {
  transactions: VMTransaction[]
  loadingState: LoadingState
  error: string | null
  loadMoreToken: string | undefined
}

const initialState: TransactionState = {
  transactions: [],
  loadingState: 'idle',
  error: null,
  loadMoreToken: undefined
}

export const transaction = createSlice({
  name: 'transaction',
  initialState,
  reducers: {
    setTransactions: (state, action: PayloadAction<VMTransaction[]>) => {
      state.transactions = action.payload
      state.loadingState = 'success'
    },
    addTransactions: (state, action: PayloadAction<VMTransaction[]>) => {
      state.transactions = state.transactions.concat(action.payload)
      state.loadingState = 'success'
    },
    setLoadingState: (state, action: PayloadAction<LoadingState>) => {
      state.loadingState = action.payload
    },
    setLoadMoreToken: (state, action: PayloadAction<string | undefined>) => {
      state.loadMoreToken = action.payload
    },
    setError: (state, action: PayloadAction<any>) => {
      state.error = action.payload
    }
  }
})
