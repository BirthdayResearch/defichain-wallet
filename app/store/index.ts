/* eslint-disable @typescript-eslint/explicit-function-return-type */
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { authentication } from './authentication'
import { block } from './block'
import { ocean } from './ocean'
import { transaction } from './transaction'
import { transactionQueue } from './transaction_queue'
import { wallet } from './wallet'

/**
 * RootState for DeFiChain Wallet App
 *
 * All state reducer in this store must be designed for global use and placed in this
 * directory as such. Reducer that are not meant to be global must not be part of
 * RootState.
 *
 * Non-global state should be managed independently within its own React Component.
 */
export function initializeStore () {
  return configureStore({
    reducer: {
      block: block.reducer,
      wallet: wallet.reducer,
      ocean: ocean.reducer,
      transactionQueue: transactionQueue.reducer,
      authentication: authentication.reducer,
      transaction: transaction.reducer
    },
    middleware: [
      ...getDefaultMiddleware({ serializableCheck: false })
    ]
  })
}

export type RootStore = ReturnType<typeof initializeStore>
export type RootState = ReturnType<RootStore['getState']>
