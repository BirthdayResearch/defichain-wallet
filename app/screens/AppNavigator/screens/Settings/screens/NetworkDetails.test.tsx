import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react-native'
import * as React from 'react'
import { NetworkDetails } from './NetworkDetails'
import { RootState } from '../../../../../store'
import { block } from '@store/block'

jest.mock('../../../../../contexts/ThemeProvider')

jest.mock('../../../../../contexts/NetworkContext', () => ({
  useNetworkContext: () => {
    return {
      network: 'Playground'
    }
  }
}))

describe('NetworkDetails', () => {
  it('<NetworkDetails /> should render components', () => {
    const initialState: Partial<RootState> = {
      block: {
        count: 2000,
        masterNodeCount: 10,
        lastSync: 'Tue Sep 14 2021 17:36:16 GMT+0530',
        connected: true,
        isPolling: true
      }
    }
    const store = configureStore({
      preloadedState: initialState,
      reducer: { block: block.reducer }
    })
    const component = (
      <Provider store={store}>
        <NetworkDetails />
      </Provider>
    )
    const rendered = render(component)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
