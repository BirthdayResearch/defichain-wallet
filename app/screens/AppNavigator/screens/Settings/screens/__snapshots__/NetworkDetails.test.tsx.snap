// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkDetails <NetworkDetails /> should render components 1`] = `
<RCTScrollView
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "rgba(245, 245, 245, 1)",
      },
    ]
  }
  testID="network_details"
>
  <View>
    <RCTScrollView
      data={
        Array [
          Object {
            "data": Array [
              Object {
                "label": "Network",
                "status": "Playground",
                "testID": "network",
              },
              Object {
                "label": "Status",
                "status": "Connected",
                "statusIcon": <View
                  style={
                    Object {
                      "backgroundColor": "rgba(16, 185, 129, 1)",
                      "borderBottomLeftRadius": 9999,
                      "borderBottomRightRadius": 9999,
                      "borderTopLeftRadius": 9999,
                      "borderTopRightRadius": 9999,
                      "height": 8,
                      "marginRight": 6,
                      "width": 8,
                    }
                  }
                  testID="status_icon_network"
                />,
                "testID": "status",
              },
            ],
            "key": "YOU ARE CURRENTLY CONNECTED TO",
          },
          Object {
            "data": Array [
              Object {
                "label": "Last synced",
                "status": "Sep 14, 9:07 pm",
                "testID": "last_sync",
              },
              Object {
                "label": "Block height",
                "status": 2000,
                "testID": "block_height",
              },
              Object {
                "label": "Total Masternodes",
                "status": 10,
                "testID": "total_masternodes",
              },
            ],
            "key": "NETWORK DETAILS",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={
        Array [
          0,
          4,
        ]
      }
      style={
        Array [
          undefined,
          Object {
            "backgroundColor": "rgba(245, 245, 245, 1)",
          },
        ]
      }
      testID="network_detail_screen_list"
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "RegularFont",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
                Array [
                  Object {
                    "color": "rgba(115, 115, 115, 1)",
                    "fontFamily": "MediumFont",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "lineHeight": 16,
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 24,
                  },
                  Object {
                    "backgroundColor": "rgba(245, 245, 245, 1)",
                    "color": "rgba(115, 115, 115, 1)",
                  },
                ],
              ]
            }
            testID="network_detail_title"
            text="YOU ARE CURRENTLY CONNECTED TO"
          >
            YOU ARE CURRENTLY CONNECTED TO
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 8,
                  "paddingTop": 16,
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 1)",
                  "borderBottomColor": "rgba(229, 229, 229, 1)",
                  "borderBottomWidth": 1,
                  "borderLeftColor": "rgba(229, 229, 229, 1)",
                  "borderRightColor": "rgba(229, 229, 229, 1)",
                  "borderTopColor": "rgba(229, 229, 229, 1)",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontFamily": "MediumFont",
                      "fontWeight": "500",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 1)",
                    },
                  ],
                ]
              }
              testID="network_label"
            >
              Network
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "color": "rgba(82, 82, 82, 1)",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "color": "rgba(82, 82, 82, 1)",
                        "fontFamily": "MediumFont",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "rgba(82, 82, 82, 1)",
                      },
                    ],
                  ]
                }
                testID="network_value"
              >
                Playground
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 8,
                  "paddingTop": 16,
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 1)",
                  "borderBottomColor": "rgba(229, 229, 229, 1)",
                  "borderBottomWidth": 1,
                  "borderLeftColor": "rgba(229, 229, 229, 1)",
                  "borderRightColor": "rgba(229, 229, 229, 1)",
                  "borderTopColor": "rgba(229, 229, 229, 1)",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontFamily": "MediumFont",
                      "fontWeight": "500",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 1)",
                    },
                  ],
                ]
              }
              testID="status_label"
            >
              Status
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "color": "rgba(82, 82, 82, 1)",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "rgba(16, 185, 129, 1)",
                    "borderBottomLeftRadius": 9999,
                    "borderBottomRightRadius": 9999,
                    "borderTopLeftRadius": 9999,
                    "borderTopRightRadius": 9999,
                    "height": 8,
                    "marginRight": 6,
                    "width": 8,
                  }
                }
                testID="status_icon_network"
              />
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "color": "rgba(82, 82, 82, 1)",
                        "fontFamily": "MediumFont",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "rgba(82, 82, 82, 1)",
                      },
                    ],
                  ]
                }
                testID="status_value"
              >
                Connected
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
        <View
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "RegularFont",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
                Array [
                  Object {
                    "color": "rgba(115, 115, 115, 1)",
                    "fontFamily": "MediumFont",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "lineHeight": 16,
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 24,
                  },
                  Object {
                    "backgroundColor": "rgba(245, 245, 245, 1)",
                    "color": "rgba(115, 115, 115, 1)",
                  },
                ],
              ]
            }
            testID="NETWORK DETAILS"
            text="NETWORK DETAILS"
          >
            NETWORK DETAILS
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 8,
                  "paddingTop": 16,
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 1)",
                  "borderBottomColor": "rgba(229, 229, 229, 1)",
                  "borderBottomWidth": 1,
                  "borderLeftColor": "rgba(229, 229, 229, 1)",
                  "borderRightColor": "rgba(229, 229, 229, 1)",
                  "borderTopColor": "rgba(229, 229, 229, 1)",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontFamily": "MediumFont",
                      "fontWeight": "500",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 1)",
                    },
                  ],
                ]
              }
              testID="last_sync_label"
            >
              Last synced
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "color": "rgba(82, 82, 82, 1)",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "color": "rgba(82, 82, 82, 1)",
                        "fontFamily": "MediumFont",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "rgba(82, 82, 82, 1)",
                      },
                    ],
                  ]
                }
                testID="last_sync_value"
              >
                Sep 14, 9:07 pm
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 8,
                  "paddingTop": 16,
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 1)",
                  "borderBottomColor": "rgba(229, 229, 229, 1)",
                  "borderBottomWidth": 1,
                  "borderLeftColor": "rgba(229, 229, 229, 1)",
                  "borderRightColor": "rgba(229, 229, 229, 1)",
                  "borderTopColor": "rgba(229, 229, 229, 1)",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontFamily": "MediumFont",
                      "fontWeight": "500",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 1)",
                    },
                  ],
                ]
              }
              testID="block_height_label"
            >
              Block height
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "color": "rgba(82, 82, 82, 1)",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "color": "rgba(82, 82, 82, 1)",
                        "fontFamily": "MediumFont",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "rgba(82, 82, 82, 1)",
                      },
                    ],
                  ]
                }
                testID="block_height_value"
              >
                2000
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 8,
                  "paddingTop": 16,
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 1)",
                  "borderBottomColor": "rgba(229, 229, 229, 1)",
                  "borderBottomWidth": 1,
                  "borderLeftColor": "rgba(229, 229, 229, 1)",
                  "borderRightColor": "rgba(229, 229, 229, 1)",
                  "borderTopColor": "rgba(229, 229, 229, 1)",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontFamily": "MediumFont",
                      "fontWeight": "500",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 1)",
                    },
                  ],
                ]
              }
              testID="total_masternodes_label"
            >
              Total Masternodes
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "color": "rgba(82, 82, 82, 1)",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "color": "rgba(82, 82, 82, 1)",
                        "fontFamily": "MediumFont",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "rgba(82, 82, 82, 1)",
                      },
                    ],
                  ]
                }
                testID="total_masternodes_value"
              >
                10
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
      </View>
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
